web/src/user_groups.ts

```diff
@@ -477,6 +477,25 @@ export function is_user_in_group(
     return false;
 }
 
+export function group_has_permission(setting_value: GroupSettingValue, group_id: number): boolean {
+    if (typeof setting_value === "number") {
+        return setting_value === group_id;
+    }
+
+    const direct_subgroup_ids = setting_value.direct_subgroups;
+    if (direct_subgroup_ids.includes(group_id)) {
+        return true;
+    }
+
+    for (const direct_subgroup_id of direct_subgroup_ids) {
+        if (is_subgroup_of_target_group(direct_subgroup_id, group_id)) {
+            return true;
+        }
+    }
+
+    return false;
+}
```

Comment:
Does `user_groups.ts` not have complete node tests? I guess that may be a thing for us to plan to address once this is finished. (Just surprised coverage didn't complain).

Reply:
IDK, maybe we should aim to have at least basic tests for all the helper functions like this for 10.0, but it can happen late in the release, and 100% coverage is not a requirement.